pennylane/devices/tests/test_measurements.py

```diff
@@ -155,8 +155,10 @@ class TestHamiltonianSupport:
     """Separate test to ensure that the device can differentiate Hamiltonian observables."""
 
     @pytest.mark.parametrize("ham_constructor", [qml.ops.Hamiltonian, qml.ops.LinearCombination])
+    @pytest.mark.filterwarnings("ignore::pennylane.PennyLaneDeprecationWarning")
```

Comment:
how come we need this now? 🤔 and what is raising the deprecation warning?

Reply:
The test case is explicitly constructing `qml.ops.Hamiltonian`, see the `ham_constructor` parametrization.

Reply:
But how come this has passed CI before this 😬 ? Shouldnt we get a fail from qml.PennylaneDeprecationWarnings?

Reply:
This is not part of PennyLane's CI. It is part of the device tests, and is used by the plugin test matrix. I don't think the plugin test matrix treats `PennyLaneDeprecationWarning` as failures, but we do see it in the warnings summary: https://github.com/PennyLaneAI/plugin-test-matrix/actions/runs/8624006838/job/23638246184

Reply:
Ah that explains it, thanks!